src/components/Combobox/Combobox.tsx

```diff
@@ -163,7 +163,7 @@ function Combobox<ItemShape>({
               'aria-readonly': true,
             }),
             ...(searchable && {
-              placeholder: value ? itemToString(value) : inputProps.placeholder,
+              placeholder: value !== undefined ? safeItemToString(value) : inputProps.placeholder,
```

Comment:
Gotcha! Interesting, that was definitely a bug, but I'm not 100% sure if that fixes it.

With your change, if value is `false`, then in L166 it would enter `safeItemToString(value)`, where the related code 

```ts
  const safeItemToString = (item: ItemShape | null) => (item ? itemToString(item) : '');
```

would return into the `placeholder` getting the value `''`. Wouldn't we want to show `"false"`?


I think the solution would be to keep using `itemToString` and not use `safeItemToString`

Reply:
You are right and also notice that if I dont update the `safeItemToString` it would display the value but dont handle the value anyway, so I changed it, have a look!